src/tree/repository.ts

```diff
@@ -5,38 +5,103 @@ import { Coordinate, PlantTreeDto } from './dto';
-  constructor(private readonly prisma: PrismaService) {}
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly MIN_ZOOM_LEVEL = 15
+  ) {}
+
+  private zoomToRadius(zoom: number): number{
+    if (zoom >= 16) return 1;
+    if (zoom >= 15) return 2;
+    return 3;
+  }
 
+  // TODO: 범위 부분 Service단으로 빼기
   async getTreesByLocation(
+    userId: string,
+    zoom: number,
     location: Coordinate,
-  ): Promise<(Restaurant & { savedBy: SavedRestaurant[] })[]> {
-    console.log('Getting restaurants by location from DB:', location);
-    // TODO: Prisma 기반?
-    return [];
+  ): Promise<Restaurant[]> {
+    if (zoom < this.MIN_ZOOM_LEVEL) return [];
+
+    const following = await this.prisma.follower.findMany({
+      where: {
+        followerId: userId,
+        status: 'ACCEPTED'
+      },
+      select: { userId: true }
+    })
+
+    const followingIds = following.map((e) => e.userId)
+    const searchTargetIds = [...followingIds, userId]
+    const [ lat, lon ] = [parseFloat(location.lat), parseFloat(location.lon)]
+
+    const radius = this.zoomToRadius(zoom)
+    const latRadian = (Math.PI * lat) / 180;
+    const degLat = 110.574; // 위도 1도의 km
+    const degLon = 111.32 * Math.cos(latRadian);
+
+    const latMin = lat - radius / degLat;
+    const latMax = lat + radius / degLat;
+    const lonMin = lon - radius / degLon;
+    const lonMax = lon + radius / degLon;
+
+    const restaurants = await this.prisma.restaurant.findMany({
+      where: {
+        latitude: { gte: latMin, lte: latMax },
+        longitude: { gte: lonMin, lte: lonMax },
+      },
+      include: {
+        savedBy: {
+          where: {
+            userId: { in: searchTargetIds }
+          },
+          include: {
+            user: { select: { id: true }},
+          },
+        }
+      }
+    })
+
+    return restaurants;
   }
 
-  async getTreeById(
+  async getTreesByRestaurantId(
     restaurantId: string,
-    userId: string,
-  ): Promise<
-    (SavedRestaurant & { user: User; restaurant: Restaurant }) | null
-  > {
-    console.log(
-      'Getting tree (SavedRestaurant) for restaurant:',
-      restaurantId,
-      'by user:',
-      userId,
-    );
-    const tree = await this.prisma.savedRestaurant.findUnique({
-      where: {
-        userId_restaurantId: { userId: userId, restaurantId: restaurantId },
+    targetUids: string[]
+  ): Promise<(SavedRestaurant & { user: User; restaurant: Restaurant })[]>{
+    return this.prisma.savedRestaurant.findMany({
+      where: { 
+        restaurantId: restaurantId, 
+        userId: { in: targetUids }
       },
       include: {
         user: true,
-        restaurant: true,
+        restaurant: true
       },
-    });
-    return tree;
+      orderBy: {
+        createdAt: 'desc'
+      }
+    })
+  }
+
+  async getTreeById(
+    ownerId: string,
+    restaurantId: string
+  ): Promise<(SavedRestaurant & { user: User; restaurant: Restaurant }) | null>{
```

Comment:
타입을 새로 명시하는게 나아보여요. 아니면 하나의 object에 다른 키로 넣어주던가요. 
```suggestion
  ): Promise<{ user: User; restaurant: Restaurant; restaurant: SavedRestaurant } | null>{
```

Reply:
{ user: User; restaurant: Restaurant; tree: SavedRestaurant; }의 새로운 타입으로 수정해 올려두겠습니다!